hudi-flink/src/test/java/org/apache/hudi/operator/utils/TestData.java

```diff
@@ -149,6 +156,51 @@ public static void checkWrittenData(
     }
   }
 
+  /**
+   * Checks the source data are written as expected.
+   *
+   * <p>Note: Replace it with the Flink reader when it is supported.
+   *
+   * @param basePath   The file base to check, should be a directory
+   * @param expected   The expected results mapping, the key should be the partition path
+   */
+  public static void checkWrittenFullData(
```

Comment:
> @lamber-ken It seems this mechanism looks more reasonable. So, a question is this method can fully replace `checkWrittenData`?

No, because the tests in StreamWriteFunctionTest can use `TestData#checkWrittenData` to check data for each checkpoint manually, like

![image](https://user-images.githubusercontent.com/20113411/107605226-885d8f00-6c6d-11eb-9142-96b1d5f1e542.png)